### DISCUSSION TITLE
Use CDK to create new resources

### DISCUSSION THREAD

**Discussion by andrea-avogadro-archiva** [aws/copilot-cli#5084] [Ideas]
Hi, is it possible to **create** new [resources ](https://aws.github.io/copilot-cli/docs/developing/addons/workload/)using CDK instead of CloudFormation? I do not need to overried anything, just create new stuff. Thanks

**Comment by dannyrandall** [comment]
Hi @andrea-avogadro-archiva! There isn't a way to do that with Addons, but you can use [CDK Overrides](https://aws.github.io/copilot-cli/docs/developing/overrides/cdk/) to _only_ make new resources, without having to modify any of the Copilot-created resources! Any resources created in the override will be added to the main CloudFormation stack.

**Reply by andrea-avogadro-archiva** [reply]
Hi @dannyrandall thanks! Any canche to add this new resources in a dedicated nested stack? Thanks

**Reply by bvtujo** [reply]
@andrea-avogadro-archiva this gets a little complex, because nested stacks can only reference an s3 URL for their templates. 

The way I see it, you could manage your addons with the cdk by creating a cdk app elsewhere, then synthesizing the template and redirecting the terminal output (the YAML stack which gets printed to stdout) to `copilot/workload/addons/cdk.yml`. 

For example:
```bash
cdk synth -a './cdk-addon/my-app.js' \
  --version-reporting false \ 
  --path-metadata false \
  --asset-metadata false \
  1>./copilot/workload/addons/cdk.yml
```

You might have to manually add some [`CfnParameter`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.CfnParameter.html)s to your CDK stack to mimic the required parameters in the addons template, and in order to avoid some CDK bootstrapping rules you'll have to customize your stack synthesizer (see [this answer on StackOverflow](https://stackoverflow.com/a/70745623/5890422)). But this should let you workaround things nicely.

I'm sorry that Copilot doesn't already behave this way. Feel free to create an issue and let others +1 it; that will help prioritize this as a feature in a future release.
